src/Filterable.ts

```diff
@@ -0,0 +1,357 @@
+  readonly partition: <U, L, A>(
+    fa: Type3<F, U, L, A>,
+    p: Predicate<A>
+  ) => Separated<Type3<F, U, L, A>, Type3<F, U, L, A>>
+  readonly filterMap: <U, L, A, B>(fa: Type3<F, U, L, A>, f: (a: A) => Option<B>) => Type3<F, U, L, B>
+  readonly filter: <U, L, A>(fa: Type3<F, U, L, A>, p: Predicate<A>) => Type3<F, U, L, A>
+}
+
+/**
+ * @typeclass
+ * @since 1.7.0
+ * @see Filterable
+ */
+export interface Filterable3C<F extends URIS3, U, L> extends Functor3C<F, U, L>, Compactable3C<F, U, L> {
+  readonly partitionMap: <RL, RR, A>(
+    fa: Type3<F, U, L, A>,
+    f: (a: A) => Either<RL, RR>
+  ) => Separated<Type3<F, U, L, RL>, Type3<F, U, L, RR>>
+  readonly partition: <A>(fa: Type3<F, U, L, A>, p: Predicate<A>) => Separated<Type3<F, U, L, A>, Type3<F, U, L, A>>
+  readonly filterMap: <A, B>(fa: Type3<F, U, L, A>, f: (a: A) => Option<B>) => Type3<F, U, L, B>
+  readonly filter: <A>(fa: Type3<F, U, L, A>, p: Predicate<A>) => Type3<F, U, L, A>
+}
+
+/**
+ * Upgrade a boolean-style predicate to an either-style predicate mapping.
+ * @function
+ * @since 1.7.0
+ */
+export const eitherBool = <A>(p: Predicate<A>) => (a: A): Either<A, A> => (p(a) ? right(a) : left(a))
+
+/**
+ * Upgrade a boolean-style predicate to a maybe-style predicate mapping.
+ * @function
+ * @since 1.7.0
+ */
+export const optionBool = <A>(p: Predicate<A>) => (a: A): Option<A> => (p(a) ? some(a) : none)
+
+/**
+ * Gets default implementation of {@link Filterable.partitionMap} using {@link Compactable.separate}
+ * @function
+ * @since 1.7.0
+ */
+export function partitionMapDefaultSeparate<F extends URIS3, U, L>(
+  F: Functor3C<F, U, L> & Pick<Compactable3C<F, U, L>, 'separate'>
+): Filterable3C<F, U, L>['partitionMap']
+export function partitionMapDefaultSeparate<F extends URIS3>(
+  F: Functor3<F> & Pick<Compactable3<F>, 'separate'>
+): Filterable3<F>['partitionMap']
+export function partitionMapDefaultSeparate<F extends URIS2, L>(
+  F: Functor2C<F, L> & Pick<Compactable2C<F, L>, 'separate'>
+): Filterable2C<F, L>['partitionMap']
+export function partitionMapDefaultSeparate<F extends URIS2>(
+  F: Functor2<F> & Pick<Compactable2<F>, 'separate'>
+): Filterable2<F>['partitionMap']
+export function partitionMapDefaultSeparate<F extends URIS>(
+  F: Functor1<F> & Pick<Compactable1<F>, 'separate'>
+): Filterable1<F>['partitionMap']
+export function partitionMapDefaultSeparate<F>(
+  F: Functor<F> & Pick<Compactable<F>, 'separate'>
+): Filterable<F>['partitionMap']
+export function partitionMapDefaultSeparate<F>(
+  F: Functor<F> & Pick<Compactable<F>, 'separate'>
+): Filterable<F>['partitionMap'] {
+  return (fa, f) => F.separate(F.map(fa, f))
+}
+
+/**
+ * Gets default implementation of {@link Filterable.partition} using {@link Filterable.partitionMap}
+ * @function
+ * @since 1.7.0
+ */
+export function partitionDefaultPartitionMap<F extends URIS3, U, L>(
+  F: Pick<Filterable3C<F, U, L>, 'URI' | 'partitionMap'>
```

Comment:
why an object here (as argument) instead of

```ts
partitionMap: Pick<Filterable3C<F, U, L>, 'partitionMap'>
```

(idem for the other default-functions)

Reply:
Object is more handy when default implementation requires more than one function (like `wither`). Also I have some problems with inference without passing `URI` in `getFilterable` of `Either` (next PR).

Reply:
> I picked this convention: get<Name>From<OtherName> which is more readabe

ok

> as well as they are a good way of reflecting relation between these typeclasses

ok

> Object is more handy when default implementation requires more than one function (like wither)

ok

> Also I have some problems with inference without passing URI in getFilterable of Either (next PR)

Ah I see, ok